lib/portus/background/sync.rb

```diff
@@ -67,9 +105,26 @@ def update_registry!(catalog)
 
         # At this point, the remaining items in the "repos" array are repos that
         # exist in the DB but not in the catalog. Remove all of them.
+        delete_maybe!(dangling_repos)
+
+        # Check that no events have happened while doing all this.
+        check_events!
+      end
+
+      # Delete the given repositories unless the configuration does not allow it.
+      def delete_maybe!(repositories)
+        return if APP_CONFIG["background"]["sync"]["sync-strategy"] == "update"
+
         portus = User.find_by(username: "portus")
-        Tag.where(repository_id: dangling_repos).find_each { |t| t.delete_and_update!(portus) }
-        Repository.where(id: dangling_repos).find_each { |r| r.delete_and_update!(portus) }
+        Tag.where(repository_id: repositories).find_each { |t| t.delete_and_update!(portus) }
```

Comment:
Rename `update_and_delete!` to just `delete!`. What exactly is being updated? When I read that I got confused. I had to read the method code to understand what's going on and I think it's ok to remove the `update` from the name method. What do you think?

Reply:
Makes sense. Even though `delete!` might conflict with some rails special method (these kinds of names are usually already picked). What do you think with `delete_by!` (since we are passing the user) ?

Reply:
Sure.